client/my-sites/post-relative-time-status/index.jsx

```diff
@@ -21,121 +20,208 @@ class PostRelativeTime extends React.PureComponent {
 		post: PropTypes.object.isRequired,
-		includeNonDraftStatuses: PropTypes.bool,
 		link: PropTypes.string,
 		target: PropTypes.string,
 		gridiconSize: PropTypes.number,
 	};
 
 	static defaultProps = {
-		includeNonDraftStatuses: false,
 		link: null,
 		target: null,
 	};
 
+	/**
+	 * Get the date to be displayed depending on the status of the post
+	 */
 	getTimestamp() {
 		switch ( this.props.post.status ) {
 			case 'new':
 				return null;
+
 			case 'draft':
-			case 'future':
+			case 'trash':
 			case 'pending':
 				return this.props.post.modified;
+
 			default:
 				return this.props.post.date;
 		}
 	}
 
-	getDisplayedTimeFromPost( post ) {
+	getDisplayedTimeForLabel() {
 		const moment = this.props.moment;
-
 		const now = moment();
+		const timestamp = moment( this.getTimestamp() );
 
-		if ( ! post ) {
-			// Placeholder text: "a few seconds ago" in English locale
-			return now.fromNow();
-		}
+		const isScheduledPost = this.props.post.status === 'future';
 
-		const { status, modified, date } = post;
-		const time = moment( includes( [ 'draft', 'pending', 'future' ], status ) ? modified : date );
-		if ( now.diff( time, 'days' ) >= 7 ) {
-			// Like "Mar 15, 2013 6:23 PM" in English locale
-			return time.format( 'lll' );
+		let displayedTime;
+		if ( isScheduledPost ) {
+			displayedTime = timestamp.calendar( null, {
+				nextDay: this.props.translate( '[tomorrow at] LT', {
+					comment: 'LT refers to time (eg. 18:00)',
+				} ),
+				sameElse: this.props.translate( 'll [at] LT', {
+					comment:
+						'll refers to date (eg. 21 Apr) for when the post will be published & LT refers to time (eg. 18:00) - and "for" is translated',
+				} ),
+			} );
+		} else {
+			if ( Math.abs( now.diff( this.getTimestamp(), 'days' ) ) < 7 ) {
+				return timestamp.fromNow();
+			}
+
+			const sameElse = this.props.translate( 'll [at] LT', {
+				comment:
+					'll refers to date (eg. 21 Apr) & LT refers to time (eg. 18:00) - "at" is translated',
+			} );
+
+			displayedTime = timestamp.calendar( null, {
+				sameElse,
+			} );
 		}
 
-		// Like "3 days ago" in English locale
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		return timestamp.diff( now, 'years' ) > 0
+			? displayedTime
+			: displayedTime.replace( timestamp.format( 'Y' ), '' );
 	}
 
 	getTimeText() {
 		const time = this.getTimestamp();
-		if ( ! time ) {
-			return null;
-		}
-
 		return (
 			<span className="post-relative-time-status__time">
-				<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
-				<time className="post-relative-time-status__time-text" dateTime={ time }>
-					{ this.getDisplayedTimeFromPost( this.props.post ) }
-				</time>
+				{ time && (
+					<>
+						<Gridicon icon="time" size={ this.props.gridiconSize || 18 } />
+						<time className="post-relative-time-status__time-text" dateTime={ time }>
+							{ this.getDisplayedTimeForLabel( true ) }
```

Comment:
`getDisplayedTimeForLabel` no longer needs arguments.
```suggestion
							{ this.getDisplayedTimeForLabel() }
```